flake.nix

```diff
@@ -6,20 +6,49 @@
   outputs = {self}: {
     __functor = self: self.lib.noSys;
 
-    lib.noSys = inputs @ {systems ? import ./systems.nix, ...}: f: let
+    lib.noSys = inputs @ {
+      systems ? import ./systems.nix,
+      config ? {},
+      ...
+    }: f: let
       systems' =
         if builtins.isPath systems || systems ? outPath
         then import systems
         else systems;
     in
-      self.lib.eachSys systems' (sys: let
+      self.lib.eachSys systems' (system: let
         f' = inputs: let
           # mapAttrs to deSys up to the same depths as in `self`
-          inputs' = builtins.mapAttrs (_: self.lib.deSys sys) (builtins.removeAttrs inputs ["self"]);
-          self' = self.lib.deSys sys (builtins.removeAttrs inputs.self ["inputs"]);
+          inputs' = builtins.mapAttrs (k: v: let
+            input = self.lib.deSys system v;
+          in
+            if
+              k
+              == "pkgs"
+              && v ? legacyPackages
+              && v.inputs == {}
```

Comment:
Why this check? (curiosity, not disagreement)

Reply:
just trying to ensure it is nixpkgs as reliably as possible